lib/ingestor-api/index.ts

```diff
@@ -153,26 +158,28 @@ export class StacIngestor extends Construct {
     dbVpc: ec2.IVpc;
     dbSecurityGroup: ec2.ISecurityGroup;
     subnetSelection: ec2.SubnetSelection;
-    ingestorCode?: IngestorCode;
-  }): PythonFunction {
-
-
-
-    const ingestorCode = props.ingestorCode || {
-      entry: `${__dirname}/runtime`,
-      index: "src/ingestor.py",
-      handler: "handler",
-    };
+    lambdaFunctionOptions?: CustomLambdaFunctionOptions;
+    lambdaAssetCode?: lambda.AssetCode;
+  }): lambda.Function {
 
-    const handler = new PythonFunction(this, "stac-ingestor", {
-      ...ingestorCode,
-      runtime: lambda.Runtime.PYTHON_3_9,
-      timeout: Duration.seconds(180),
-      environment: { DB_SECRET_ARN: props.dbSecret.secretArn, ...props.env },
+    
+    const handler = new lambda.Function(this, "stac-ingestor", {
+      ...props.lambdaFunctionOptions ?? {
+        runtime: lambda.Runtime.PYTHON_3_9,
+        handler: "handler.handler",
+        memorySize: 2048,
+        logRetention: aws_logs.RetentionDays.ONE_WEEK,
+        timeout: Duration.seconds(180)
+      },
+      code: props.lambdaAssetCode ?? lambda.Code.fromDockerBuild(__dirname, {
+        file: "runtime/Dockerfile",
+        buildArgs: { PYTHON_VERSION: '3.9' },
```

Comment:
It doesn't appear that any of your updated `Dockerfile`'s are actually leveraging `buildArgs`.

Reply:
@sharkinsspatial I think this is because now the user has to pass `aws_cdk.aws_lambda.Code` https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_lambda/Code.html so we can't really just pass build-arg as before

Reply:
@vincentsarago I might be misreading the TypeScript here but I think that construct users can optionally specify their own [lambda.assetCode](https://github.com/developmentseed/eoapi-cdk/blob/eb20644460d8f3f59339eb48bb3239cabea83860/lib/stac-api/index.ts#L31) or the construct will use `lambda.Code.fromDockerBuild` with the local runtime and `Dockerfile`.  Currently, the base image in the `Dockerfile` is not using the `buildArgs` that are passed.  They should be so that the base image can be specified dynamically to use the Python version set in the `runtime` parameter.

Reply:
I agree that user should still be able to pass custom buildArgs even if we switch to full AssetCode

Reply:
@sharkinsspatial

> It doesn't appear that any of your updated Dockerfile's are actually leveraging buildArgs.

This is correct, and I fixed this mistake last friday but hadn't pushed . That said, your idea here : 

> dynamically specifying the base Lambda image's Python version

That would be great, but only if it's easy to use. There seems to be just that [exposed by the CDK](https://github.com/aws/aws-cdk/issues/26372). Could try that.

Reply:
Unfortunately, like mentioned by an aws-cdk maintainer in [this issue](https://github.com/aws/aws-cdk/issues/26372), this feature seems to be supported for `lambda.DockerImageFunction` only.

Reply:
> I agree that user should still be able to pass custom buildArgs even if we switch to full AssetCode

that's going to be the case now, see https://github.com/developmentseed/eoapi-cdk/commit/dbcada8cf65e3264a71144c660e9a688db9f1e59